java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -310,6 +311,52 @@ public BaseDeviceMemoryBuffer getDeviceBufferFor(BufferType type) {
     return srcBuffer;
   }
 
+  /**
+   * Create a ColumnVector from the Apache Arrow byte buffers passed in.
+   * Any of the buffers not used for that datatype should be set to null.
+   * The buffers are expected to be off heap buffers, but if they are not,
+   * it will handle copying them to direct byte buffers.
+   * This only supports primitive types and Strings, Decimals and nested types
+   * such as list and struct are not supported.
+   * @param type - type of the column
+   * @param numRows - Number of rows in the arrow column
+   * @param nullCount - Null count
+   * @param data - ByteBuffer of the Arrow data buffer
+   * @param validity - ByteBuffer of the Arrow validity buffer
+   * @param offsets - ByteBuffer of the Arrow offsets buffer
+   * @return - new ColumnVector
+   */
+  public static ColumnVector fromArrow(
+      DType type,
+      long numRows,
+      long nullCount,
+      ByteBuffer data,
+      ByteBuffer validity,
+      ByteBuffer offsets) {
+    ByteBuffer dataBuffer = data;
+    if (dataBuffer != null && !dataBuffer.isDirect()) {
```

Comment:
Nit: would be nice to have a private utility method, e.g.: `bufferAsDirect` or something similar, that reduces the boilerplate code here.

Reply:
added